gb-cpu/src/memory.rs

```diff
@@ -0,0 +1,103 @@
+pub mod area;
+mod consts;
+
+use crate::error::Error;
+
+use crate::bus::Bus;
+use area::rom::{Mbc, NoMbc};
+use area::{Area, Wram};
+
+type Rom = Box<dyn Bus<usize, Item = u8, Result = Result<(), Error>, Data = u8>>;
+
+pub struct Memory {
+    pub wram: Wram,
+    pub rom: Rom,
+}
+
+impl Bus<u16> for Memory {
+    type Item = Result<u8, Error>;
+    type Result = Result<(), Error>;
+    type Data = u8;
```

Comment:
these type are not used in the implementation, Consider using/removing them

Reply:
use the type that you've defined It'll be more clear for me

Reply:
Great point, i'll update that